corehq/apps/app_manager/suite_xml/sections/entries.py

```diff
@@ -125,14 +126,10 @@ def get_registry_nodeset_xpath(case_type, filter_xpath='', additional_types=None
 
     @staticmethod
     def _get_nodeset_xpath(instance_name, root_element, case_type, filter_xpath='', additional_types=None):
-        if additional_types:
-            additional_types = set(additional_types) - {case_type}
-            case_type_filter = " or ".join([
-                "@case_type='{case_type}'".format(case_type=case_type)
-                for case_type in [case_type] + sorted(additional_types)
-            ])
-        else:
-            case_type_filter = "@case_type='{case_type}'".format(case_type=case_type)
+        case_type_filter = " or ".join([
+            "@case_type='{case_type}'".format(case_type=case_type)
+            for case_type in get_ordered_case_types(case_type, additional_types)
+        ])
```

Comment:
the branching her is unnecessary since the function will return a single case type if there are no additional case types.